### PR TITLE
Sketcher: ViewProviderSketch remove unecessary updateColor and applyCursor

### PR DESCRIPTION
**Pull request by PaddleStroke**
Remove unecessary updateColor and applyCursor.

While doing https://github.com/FreeCAD/FreeCAD/pull/15160 I noticed that some applyCursor and updateColor were unecessary.

### CODE REVIEW

File: src/Mod/Sketcher/Gui/ViewProviderSketch.cpp [hunk 1]
```diff
@@ -2092,38 +2095,32 @@ void ViewProviderSketch::onSelectionChanged(const Gui::SelectionChanges& msg)
...
-                        this->updateColor();
                     }
                     else if (shapetype.size() > 10 && shapetype.substr(0, 10) == "Constraint") {
                         int ConstrId =
                             Sketcher::PropertyConstraintList::getIndexFromConstraintName(shapetype);
                         selection.SelConstraintSet.insert(ConstrId);
                         editCoinManager->drawConstraintIcons();
-                        this->updateColor();
                     }
+                    updateColor();
```
Comment (PaddleStroke) [inline src/Mod/Sketcher/Gui/ViewProviderSketch.cpp:2123]: All cases are covered here. So updateColor can be outside.

File: src/Mod/Sketcher/Gui/ViewProviderSketch.cpp [hunk 2]
```diff
@@ -2139,40 +2136,34 @@ void ViewProviderSketch::onSelectionChanged(const Gui::SelectionChanges& msg)
...
                         }
                         else if (shapetype.size() > 10 && shapetype.substr(0, 10) == "Constraint") {
                             int ConstrId =
                                 Sketcher::PropertyConstraintList::getIndexFromConstraintName(
                                     shapetype);
                             selection.SelConstraintSet.erase(ConstrId);
                             editCoinManager->drawConstraintIcons();
-                            this->updateColor();
                         }
+                        updateColor();
```
Comment (PaddleStroke) [inline src/Mod/Sketcher/Gui/ViewProviderSketch.cpp:2166]: Same as above.

File: src/Mod/Sketcher/Gui/ViewProviderSketch.cpp [hunk 3]
```diff
@@ -2204,37 +2195,22 @@ void ViewProviderSketch::onSelectionChanged(const Gui::SelectionChanges& msg)
                         int GeoId = std::atoi(&shapetype[4]) - 1;
                         resetPreselectPoint();
                         preselection.PreselectCurve = GeoId;
-
-                        if (sketchHandler)
-                            sketchHandler->applyCursor();
-                        this->updateColor();
```
Comment (PaddleStroke) [inline src/Mod/Sketcher/Gui/ViewProviderSketch.cpp:2210]: This is already done in mouseMove

File: src/Mod/Sketcher/Gui/ViewProviderSketch.cpp [hunk 4]
```diff
@@ -2204,37 +2195,22 @@ void ViewProviderSketch::onSelectionChanged(const Gui::SelectionChanges& msg)
...
                     }
                     else if (shapetype.size() > 12 && shapetype.substr(0, 12) == "ExternalEdge") {
                         int GeoId = std::atoi(&shapetype[12]) - 1;
                         GeoId = -GeoId - 3;
                         resetPreselectPoint();
                         preselection.PreselectCurve = GeoId;
-
-                        if (sketchHandler)
-                            sketchHandler->applyCursor();
-                        this->updateColor();
```
Comment (PaddleStroke) [inline src/Mod/Sketcher/Gui/ViewProviderSketch.cpp:2220]: same

File: src/Mod/Sketcher/Gui/ViewProviderSketch.cpp [hunk 5]
```diff
@@ -2204,37 +2195,22 @@ void ViewProviderSketch::onSelectionChanged(const Gui::SelectionChanges& msg)
...
                     }
                     else if (shapetype.size() > 6 && shapetype.substr(0, 6) == "Vertex") {
                         int PtIndex = std::atoi(&shapetype[6]) - 1;
                         setPreselectPoint(PtIndex);
-
-                        if (sketchHandler)
-                            sketchHandler->applyCursor();
-                        this->updateColor();
```
Comment (PaddleStroke) [inline src/Mod/Sketcher/Gui/ViewProviderSketch.cpp:2228]: same

File: src/Mod/Sketcher/Gui/ViewProviderSketch.cpp [hunk 6]
```diff
@@ -2260,8 +2236,6 @@ bool ViewProviderSketch::detectAndShowPreselection(SoPickedPoint* Point, const S
             if (accepted) {
                 setPreselectPoint(result.PointIndex);
 
-                if (sketchHandler)
-                    sketchHandler->applyCursor();
```
Comment (PaddleStroke) [inline src/Mod/Sketcher/Gui/ViewProviderSketch.cpp:2264]: This is moved to mouseMove so that it's done only once.

**Comment by PaddleStroke** [comment]
This is a refactor and does not change anything so perhaps post 1.0

File: src/Mod/Sketcher/Gui/ViewProviderSketch.cpp (hunk 3 above)
Comment (hyarion) [reply src/Mod/Sketcher/Gui/ViewProviderSketch.cpp:2210]: It's done here: https://github.com/FreeCAD/FreeCAD/blob/ea6c4ea60fd998429108b864ea9a56e21b60df76/src/Mod/Sketcher/Gui/ViewProviderSketch.cpp#L741
which is called here:
https://github.com/FreeCAD/FreeCAD/blob/f9ba94c70b139ece6ba5ffba47632d91b5c9c645/src/Mod/Sketcher/Gui/DrawSketchController.h#L567

So while it's not done _in_ `mouseMove`, it is called after.

**Review by hyarion — APPROVED** [review]
Looks correct to me

**Pull request merged**